maestro-app/src/test/java/bio/overture/maestro/app/infra/adapter/inbound/messaging/song/SongAnalysisStreamListenerTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2019. Ontario Institute for Cancer Research
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU Affero General Public License as
+ *   published by the Free Software Foundation, either version 3 of the
+ *   License, or (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU Affero General Public License for more details.
+ *
+ *   You should have received a copy of the GNU Affero General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package bio.overture.maestro.app.infra.adapter.inbound.messaging.song;
+
+import bio.overture.maestro.app.infra.adapter.inbound.messaging.MessagingConfig;
+import bio.overture.maestro.domain.api.Indexer;
+import bio.overture.maestro.domain.api.message.AnalysisIdentifier;
+import bio.overture.maestro.domain.api.message.IndexAnalysisCommand;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.Ignore;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.ContextConfiguration;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.*;
+
+
+@Disabled
+@Ignore
```

Comment:
I thought I could get this in with this PR but this test seems to need more debugging .. so I ignored it for now until I figure it out. it's irrelevant to this change